File: iosApp/iosApp/Services/CultServiceAssembler.swift
// Created by web3d3v on 04/06/2022.
// Copyright (c) 2022 Sons Of Crypto.
// SPDX-License-Identifier: MIT

import Foundation
import web3walletcore

final class CultServiceAssembler: AssemblerComponent {

    func register(to registry: AssemblerRegistry) {

        registry.register(scope: .singleton) { resolver -> CultService in
            DefaultCultService(
                store: KeyValueStore(name: "\(CultService.self)")
            )
        }
    }
}

File: androidApp/src/main/java/com/sonsofcrypto/web3wallet/android/services/CultServiceAssembler.kt
package com.sonsofcrypto.web3wallet.android.services

import com.sonsofcrypto.web3lib.keyValueStore.KeyValueStore
import com.sonsofcrypto.web3wallet.android.common.AssemblerComponent
import com.sonsofcrypto.web3wallet.android.common.AssemblerRegistry
import com.sonsofcrypto.web3wallet.android.common.AssemblerRegistryScope
import com.sonsofcrypto.web3walletcore.services.cult.CultService
import com.sonsofcrypto.web3walletcore.services.cult.DefaultCultService
import smartadapter.internal.extension.name

class CultServiceAssembler: AssemblerComponent {

    override fun register(to: AssemblerRegistry) {

        to.register(CultService::class.name, AssemblerRegistryScope.INSTANCE) {
            DefaultCultService(
                it.resolve("${KeyValueStore::class.name}.CultService"),
            )
        }
    }
}